src/test/scala/cromwell/util/TryUtilSpec.scala

```diff
@@ -72,4 +79,53 @@ class TryUtilSpec extends FlatSpec with Matchers with MockitoSugar {
     value.failed.get.isInstanceOf[CromwellFatalException] shouldBe true
     work.counter shouldBe 3
   }
+
+  it should "Not count transient errors against the max limit" in {
+   
+    def isTransient(t: Throwable) = t match {
+      case e: TransientException => true
+      case _ => false
+    }
+
+    val work = new MockWork
+    val value = TryUtil.retryBlock(
+      fn = work.failNTimes(5, 0),
+      backoff = backoff,
+      retryLimit = Some(5),
+      logger = logger,
+      failMessage = Some(s"failed attempt (on purpose)"),
+      isTransient = isTransient
+    )
+
+    value.isFailure shouldEqual true
+    work.counter shouldBe 0
+
+    val work2 = new MockWork
+    val value2 = TryUtil.retryBlock(
+      fn = work2.failNTimes(5, 1),
+      backoff = backoff,
+      retryLimit = Some(5),
+      logger = logger,
+      failMessage = Some(s"failed attempt (on purpose)"),
+      isTransient = isTransient
+    )
+
+      value2.isFailure shouldEqual false
+      value2 shouldEqual Success(9)
+      work2.counter shouldBe 0
+
+    val work3 = new MockWork
+    val value3 = TryUtil.retryBlock(
+      fn = work3.failNTimes(6, 4),
+      backoff = backoff,
+      retryLimit = Some(5),
+      logger = logger,
+      failMessage = Some(s"failed attempt (on purpose)"),
+      isTransient = isTransient
+    )
+
+    value3.isFailure shouldEqual false
+    value3 shouldEqual Success(9)
+    work3.counter shouldBe 0
+  }
```

Comment:
There is already one :

```
it should "Fail if it hits a fatal exception" in {
     val work = new MockWork
     val value = TryUtil.retryBlock(
       fn = work.failNTimes(4),
       retryLimit = Some(4),
       backoff = backoff,
       logger = logger,
       failMessage = Some(s"failed attempt (on purpose)"),
       isFatal = (t: Throwable) => t.isInstanceOf[IllegalArgumentException]
     )
     value.isFailure shouldEqual true
      value.failed.get.isInstanceOf[CromwellFatalException] shouldBe true
      work.counter shouldBe 3
    }
```

Maybe one that combines `isFatal` + `isTransient` would be good to though.